samza-core/src/main/java/org/apache/samza/operators/impl/OperatorImpl.java

```diff
@@ -64,4 +84,7 @@ void propagateResult(RM outputMessage, MessageCollector collector, TaskCoordinat
     nextOperators.forEach(sub -> sub.onNext(outputMessage, collector, coordinator));
   }
 
+  private void propagateTimer(MessageCollector collector, TaskCoordinator coordinator) {
```

Comment:
There are 2 possible approaches here:
1. Have a single timer queue for the entire operatorGraph.
2. Have per-instance queues for each of the individual `operatorImpl`s in the operatorGraph. (as is being done currently by this patch)

The net number of callbacks is exactly equivalent. Assuming `k` operators, Having one queue with size `N` is equivalent to `k` queues with size `N/k`. There maybe trivial gains in iterating over the graph (but, that's just a no-op traversal or two)

Approach [2] also has a nice property that there's no shared state across operator impls. (queues are local to the operator).

(Had a discussion and explained this with Yi offline)